src/components/BatchClassComponents/PathwayCourseBatchEnroll1.js

```diff
@@ -0,0 +1,133 @@
+    setUpcomingBatchesOpen(true);
+  };
+  const handleUpcomingBatchesClickClose = () => {
+    setUpcomingBatchesOpen(false);
+  };
+
+  return BatchData ? (
+    <>
+      <Container maxWidth="lg">
+        <Box align="right" mt={1} maxWidth={500} mb={10}>
+          <Card elevation={2} pl={10}>
+            <CardContent>
+              <Typography gutterBottom variant="h5" align="start">
+                {BatchData?.title}
+              </Typography>
+              <Typography
+                variant="body1"
+                mb={1}
+                style={{
+                  display: "flex",
+                  padding: "10px 0",
+                }}
+              >
+                <img
+                  className={classes.icons}
+                  src={require("./assets/calender.svg")}
+                  alt="Students Img"
+                />
+                From {format(BatchData?.start_time, "dd MMM yy")} -{" "}
+                {format(BatchData?.end_time, "dd MMM yy")}
+              </Typography>
+              <Typography
+                variant="body1"
+                mb={1}
+                style={{
+                  display: "flex",
+                }}
+              >
+                <img
+                  className={classes.icons}
+                  src={require("./assets/degree.svg")}
+                  alt="Students Img"
+                />
+                Access to live classes
+              </Typography>
+              <Button variant="contained" onClick={handleClickOpen} fullWidth>
+                Enroll Batch
+              </Button>
+              <AlertDialog
+                open={open}
+                close={close}
+                title={BatchData?.title}
+                start_time={BatchData?.start_time}
+                end_time={BatchData?.end_time}
+                id={BatchData?.id}
+                registerAll={true}
+                type="batch"
+              />
+              <Typography
+                style={{ display: "flex" }}
+                mt={2}
+                align="start"
+                variant="body2"
+              >
+                Can’t start on {format(BatchData?.start_time, "dd MMM yy")}
+                {" ? "}
+                <section
+                  className={classes.link}
+                  onClick={handleUpcomingBatchesClickOpen}
+                  style={{
+                    cursor: "pointer",
+                  }}
+                >
+                  {"  "} &nbsp;
+                  <b>Check out our other batches</b>
+                </section>
+                <CheckMoreBatches
+                  open={upcomingBatchesOpen}
+                  handleUpcomingBatchesClickOpen={
+                    handleUpcomingBatchesClickOpen
+                  }
+                  handleUpcomingBatchesClickClose={
+                    handleUpcomingBatchesClickClose
+                  }
+                  upcomingBatchesData={upcomingBatchesData}
+                />
```

Comment:
I haven't fully gone through this, but what's the difference between `PathwayCourseBatchEnroll1` and `PathwayCourseBatchEnroll2` and can we factor out duplicate code from these Components?  For example both components make use of `CheckMoreBatches` and `AlertDialog` with the same property values.

Reply:
the design is completely diffrent like 
PathwayCourseBatchEnroll1  is
![image](https://user-images.githubusercontent.com/84696578/175467176-94cb1497-b84a-4131-b0e9-7046ad23c70c.png)
PathwayCourseBatchEnroll2 is
![image](https://user-images.githubusercontent.com/84696578/175467216-52d54ec9-bed5-48e7-ba8f-33fa4e21e8b3.png)
we can convert It into one component but will increases conditioning so kept it different.

Reply:
@kartiks26 Is it just design because what about how enrolling is handled?
https://github.com/navgurukul/bhanwari-devi/blob/16b0f1892021cb19ec0e0014f7bafa74f1ae2176/src/components/BatchClassComponents/PathwayCourseBatchEnroll2.js#L21-L23
versus https://github.com/navgurukul/bhanwari-devi/blob/16b0f1892021cb19ec0e0014f7bafa74f1ae2176/src/components/BatchClassComponents/PathwayCourseBatchEnroll1.js#L25-L31
And if it is only design (or even if they share partial functionality), we can still have 2 components but should be factoring out the functionality that's common.  You don't want to have to modify both components with the same changes when common functionality changes.  You've already partially done this when you have them both include say a `<CheckMoreBatches>` component, but we want to do this to the maximum extent possible to avoid introducing potential discrepancies between the Components when we make changes that we want them to both share.

Reply:
ohkk will merge it into one